pkg/asset/installconfig/aws/validation.go

```diff
@@ -165,3 +175,41 @@ func validateDuplicateSubnetZones(fldPath *field.Path, subnets map[string]Subnet
 	}
 	return allErrs
 }
+
+func validateServiceEndpoints(fldPath *field.Path, region string, services []awstypes.ServiceEndpoint) error {
+	if isAWSSDKRegion(region) {
+		return nil
+	}
+
+	resolver := newAWSResolver(region, services)
+	var errs []error
+	for _, service := range requiredServices {
+		_, err := resolver.EndpointFor(service, region, endpoints.StrictMatchingOption)
+		if err != nil {
+			errs = append(errs, errors.Wrapf(err, "failed to find endpoint for service %q", service))
+		}
+	}
+	return utilerrors.NewAggregate(errs)
+}
+
+func isAWSSDKRegion(region string) bool {
+	for _, partition := range endpoints.DefaultPartitions() {
+		for _, partitionRegion := range partition.Regions() {
+			partitionRegion := partitionRegion
```

Comment:
nit: based on the reference in https://github.com/openshift/installer/pull/3277#discussion_r398057772 it seems like this pattern is useful for goroutines but seems like a noop here.

Reply:
so every time a use the value of range to move out of scope, i.e. range and collect into some other variable which is stored or returned and lives beyond the function, i tend to use this.

but since here it's used only for checking, i have removed it as that's not necessary 6ced6ad35d604cc6d81522c2b40fecd1fa944bad